website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+var $ = require('jquery');
+var m = require('mithril');
+var moment = require('moment');
+var $osf = require('js/osfHelpers');
+var LogText = require('js/logTextParser');
+var pips = require('sliderPips');
+
+var xhrconfig = function (xhr) {
+    xhr.withCredentials = true;
+    xhr.setRequestHeader('Content-Type', 'application/vnd.api+json');
+    xhr.setRequestHeader('Accept', 'application/vnd.api+json; ext=bulk');
+
+};
+
+var LogWrap = {
+    controller: function(args){
+        var self = this;
+        self.userId = args.userId;
+        self.activityLogs = m.prop();
+        self.eventFilter = false;
+        self.dateEnd = moment.utc();
+        self.dateBegin = moment.utc();
+        self.today = moment.utc();
+        self.sixMonthsAgo = moment.utc().subtract(6, 'months');
+        self.page = 1;
+        self.cache = [];
+        self.loading = false;
+        self.div = 8.64e+7;
+        self.canvasHeight = 40;
+        self.totalEvents = 0;
+        self.eventNumbers = {};
+
+        self.getLogs = function(init, reset, update) {
+            if (!(init || reset || update)  && self.cache[self.page - 1]){
+                self.activityLogs(self.cache[self.page - 1]);
+                if (!self.cache[self.page] && self.page < self.lastPage){
+                    self.page = self.page + 1;
+                    self.getLogs(false, false, true);
+                    self.page = self.page - 1;
+                }
+                return;
+            }
+            var query = {
+                'embed': ['nodes', 'user', 'linked_node', 'template_node'],
+                'page': ((self.page/2) | 0) + 1,
+                'page[size]': 20
+            };
+            if (self.eventFilter) {
+                query['filter[action]'] = self.eventFilter;
+            }
+            if (reset || init) {
+                query.aggregates = 1;
+            }
+            if (!init) {
+                query['filter[date][lte]'] = self.dateEnd.toISOString();
+                query['filter[date][gte]'] = self.dateBegin.toISOString();
+            }
+            var url = $osf.apiV2Url('users/' + self.userId + '/logs/', { query : query});
```

Comment:
Is there not a way to get the users logs without having to pass the user id through contextVars? Like ```v2/me/logs```?